crates/typst/src/model/selector.rs

```diff
@@ -55,7 +55,7 @@ pub enum Selector {
     ///
     /// If there is a dictionary, only elements with the fields from the
     /// dictionary match.
-    Elem(Element, Option<Dict>),
+    Elem(ElementData, Option<Vec<(u8, Value)>>),
```

Comment:
We've discussed it before, but I just wanted to add that I think this `u8` based selector approach with early validation is really clever.

Reply:
Thank you <3